### DISCUSSION TITLE
Offboard control law.

### DISCUSSION THREAD

**Discussion by Carmol1** [USC-ACTLab/crazyswarm#682] [Q&A]
HI.

I am trying to program my own controller onboard by overwriting the controller_mellinger.c, since as far as I know the crazyswarm only allows the use of the Mellinger and PID controllers and, on the other hand, I want to make use of a formation controller, but in the bitcraze forum they told me that it is no possible to make use of the states of other crazyflie in a controller onboard, for this motive, I would like to know if can you help me to implemt an offboard controller?. So far I have written a  ROS subscriber node by using Python (the topic by using is /tf), to get the information from the Motive sofware and use it in the control law.

As additional data I get like control inputs the torque, for such reason, I do not know if it is possible to use cmdVel(roll, pitch, yawrate, thrust) or exist other command?

Regards.

**Comment by whoenig** [comment]
- It is possible to get the neighbor's position onboard the Crazyflie. One example for that is the collision avoidance module that @jpreiss contributed. If position (and potentially numerically estimated velocity) suffice, you could implement your controller directly on-board.
- Off-board, you can use tf to get neighbors position. You can then use cmdVel, cmdFullState or others, to control the Crazyflies (depending on what the output of your control law is).
